src/interfaces/DownloadFileAction.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IWidgetApiRequest, IWidgetApiRequestData } from "./IWidgetApiRequest";
+import { IWidgetApiResponseData } from "./IWidgetApiResponse";
+import { WidgetApiFromWidgetAction } from "./WidgetApiAction";
+
+export interface IDownloadFileActionFromWidgetRequestData
+  extends IWidgetApiRequestData {
+  content_uri: string;  // eslint-disable-line camelcase
+}
+
+export interface IDownloadFileActionFromWidgetActionRequest
+  extends IWidgetApiRequest {
+  action: WidgetApiFromWidgetAction.MSC4039DownloadFileAction;
+  data: IDownloadFileActionFromWidgetRequestData;
+}
+
+export interface IDownloadFileActionFromWidgetResponseData
+  extends IWidgetApiResponseData {
+  file: XMLHttpRequestBodyInit;
```

Comment:
Also, another comment that's really on the MSC, but... why `XMLHttpRequestBodyInit`? This seems like a bit of a weird union type defined in the fetch API for... actually I'm not even sure why, it doesn't seem like it's referenced. The react-sdk impl just sends a blob. If something sends ones of the other types, how would it go about determining what it's actually got? `instanceof` in js can be... special.

Reply:
The explanations I can offer is, quoting from the MSC:

To your first question:
> `response.file` mirrors `data.file` of the `upload_file` action and is a `XMLHttpRequestBodyInit` that is supported as a data type in the `postMessage` API.

To be consistent, would you think that we would also have to rename the upload's data.file? We would then need to spend some thoughts on backwards compatibility.


To the second one:
`XMLHttpRequestBodyInit` is reused again here because it was already used in https://github.com/matrix-org/matrix-widget-api/blob/3c9543cbe4255869953d1b79d513de2eaf87033e/src/interfaces/UploadFileAction.ts#L23.

Again, to be consistent we would need to change it there too and manage version incompatibilities.

Reply:
Well, IMO renaming file on the upload API would be clearer, but that's for the MSC discussion anyway. If we think the right way to go is to change upload too, then we'd need a new version of the MSC anyway, so we may as well implement download as the MSC is now and then change both at the same time, at a later date?